### DISCUSSION TITLE
Some ideas

### DISCUSSION THREAD

**Discussion by gege83var** [capsule-corp-ternoa/ternoa-subql#115] [Ideas]
Hi everyone,

-Is it possible to add a way to use id as integer (or modify the order by id), cause for NFT id (mainly, but with all technically), the order by id consider "919"<"92"<"920" instead of 92<919<920 and it's not convenient.
For some NFT, the order by created timestamp doens't work too cause the timestamp is the same for all the NFT in the same collection (if they are mint at the same time of course)

-Is it possible to add a way to order by a distinct count ? For exemple :

```
query nombreDeCometParWallet{
    nftEntities(
        offset:0
    filter:{
            collectionId: {equalTo: "82"}
    }
  ){
        groupedAggregates(
      groupBy: OWNER
    ){
      keys
            distinctCount{
                nftId
      }
    }
  }
}
```

In that I would like order by distinctCount(nftId) but I don't know if it's already possible or if it's possible to add it :)

-Is it possible to increase the number of output possible with the indexer ? (capped at 100 for now)

And thanks guys for your work <3

**Comment by ipapandinas** [comment]
hi @gege83var - thank you for opening this discussion, I am moving it to the [Indexer](https://github.com/capsule-corp-ternoa/ternoa-subql) repository since your query is related to `nftEntities`.

Here are my responses to your listed questions:
- You are right, storing ids as strings disturbs order; we are considering change the type to number.
- No it is not possible.
- Request responses are limited to 100 items to avoid overloading. The solution here is to paginate your queries using offset and first filters. An example is available in our old marketplace api [here](https://github.com/capsule-corp-ternoa/nft-marketplace-api/blob/3a49eef9411dc1de3f4287985530015fcb570f5a/src/api/services/gqlQueriesBuilder.ts#L28).
